### PR TITLE
Make Google Private Key accessible from the ENV. 

### PR DESCRIPTION
**Pull request by orangewolf**
This makes K8S deploys and dynamic configuration for Hyku easier.

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/services/hyrax/analytics/google.rb [hunk 1]
```diff
@@ -9,12 +9,15 @@ module Google
       extend ActiveSupport::Concern
       # rubocop:disable Metrics/BlockLength
       class_methods do
-        # Loads configuration options from config/analytics.yml. Expected structure:
+        # Loads configuration options from config/analytics.yml. You only need PRIVATE_KEY_PATH or
+        # PRIVA_KEY_VALUE. VALUE htakes precedence.
```
Comment (alishaevn) [inline app/services/hyrax/analytics/google.rb:13]:
```suggestion
        # PRIVATE_KEY_VALUE. VALUE takes precedence.
```

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: app/services/hyrax/analytics/google.rb [hunk 2]
```diff
@@ -50,17 +54,13 @@ def initialize(config)
           # @return [Boolean] are all the required values present?
           def valid?
             config_keys = @config.keys
+            return false unless config_keys.include?('privkey_value') || config_keys.include?('privkey_path')
             REQUIRED_KEYS.all? { |required| config_keys.include?(required) }
```
Comment (alishaevn) [inline app/services/hyrax/analytics/google.rb:58]:
```suggestion
            REQUIRED_KEYS.all? { |required| config_keys.include?(required) && @config[required].present? }
```

Comment (alishaevn) [reply app/services/hyrax/analytics/google.rb:58]: make sure that the values actually exist and aren't just empty strings.

**Review by alishaevn — CHANGES_REQUESTED** [review]

File: spec/lib/hyrax/analytics_spec.rb [hunk 3]
```diff
@@ -2,11 +2,12 @@
...
     ENV['GOOGLE_ANALYTICS_ID'] = 'UA-XXXXXXXX'
-    ENV['GOOGLE_OAUTH_APP_NAME'] = "My App Name"
-    ENV['GOOGLE_OAUTH_APP_VERSION'] = "0.0.1"
-    ENV['GOOGLE_OAUTH_PRIVATE_KEY_PATH'] = "/tmp/privkey.p12"
-    ENV['GOOGLE_OAUTH_PRIVATE_KEY_SECRET'] = "s00pers3kr1t"
-    ENV['GOOGLE_OAUTH_CLIENT_EMAIL'] = "oauth@example.org"
+    ENV['GOOGLE_OAUTH_APP_NAME'] = 'My App Name'
+    ENV['GOOGLE_OAUTH_APP_VERSION'] = '0.0.1'
+    ENV['GOOGLE_OAUTH_PRIVATE_KEY_PATH'] = '/tmp/privkey.p12'
+    ENV['GOOGLE_OAUTH_PRIVATE_KEY_VALUE'] = ''
```
Comment (alishaevn) [inline spec/lib/hyrax/analytics_spec.rb:8]: added this line. the rest of the env changes are unifying them with single quotes.

**Comment by alishaevn** [comment]
each pipeline is failing for the same 2 specs in "spec/lib/hyrax/analytics_spec.rb". those specs are passing locally for me however. may need to check in on the state of circle ci tomorrow per @dlpierce's [comment](https://github.com/samvera/hyrax/pull/5987#issuecomment-1557408157).

image

**Review by aprilrieger — APPROVED** [review]

**Pull request merged**